packages/datadog-plugin-sharedb/src/index.js

```diff
@@ -1,35 +1,55 @@
 'use strict'
 
-/**
- * @description The enum values in this map are not exposed from ShareDB, so the keys are hard-coded here.
- * The values were derived from: https://github.com/share/sharedb/blob/master/lib/client/connection.js#L196
- */
-const READABLE_ACTION_NAMES = {
-  hs: 'handshake',
-  qf: 'query-fetch',
-  qs: 'query-subscribe',
-  qu: 'query-unsubscribe',
-  bf: 'bulk-fetch',
-  bs: 'bulk-subscribe',
-  bu: 'bulk-unsubscribe',
-  f: 'fetch',
-  s: 'subscribe',
-  u: 'unsubscribe',
-  op: 'op',
-  nf: 'snapshot-fetch',
-  nt: 'snapshot-fetch-by-ts',
-  p: 'presence-broadcast',
-  pr: 'presence-request',
-  ps: 'presence-subscribe',
-  pu: 'presence-unsubscribe'
-}
+const Plugin = require('../../dd-trace/src/plugins/plugin')
+const { storage } = require('../../datadog-core')
+const analyticsSampler = require('../../dd-trace/src/analytics_sampler')
+
+class SharedbPlugin extends Plugin {
+  static get name () {
+    return 'sharedb'
+  }
+
+  constructor (...args) {
+    super(...args)
+
+    this.addSub(`apm:sharedb:_handleMessage:start`, ({ actionName, request }) => {
+      const store = storage.getStore()
+      const childOf = store ? store.span : store
+      const span = this.tracer.startSpan('sharedb.request', {
+        childOf,
+        tags: {
+          'service.name': this.config.service || this.tracer._service,
+          'span.kind': 'server',
+          'sharedb.action': actionName,
+          'resource.name': getReadableResourceName(actionName, request.c, request.q)
+        }
+      })
+
+      if (this.config.hooks && this.config.hooks.receive) {
+        this.config.hooks.receive(span, request)
+      }
 
-function getReadableActionName (action) {
-  const actionName = READABLE_ACTION_NAMES[action]
-  if (actionName === undefined) {
-    return action
+      analyticsSampler.sample(span, this.config.measured)
```

Comment:
Is this needed? The previous code doesn't seem to have used analytics sampler. Am I missing something here?